Update sidebar styling

magland opened the pull request:
Before / After

![image](https://github.com/user-attachments/assets/914108dd-e8d9-433d-9741-0fcf9e7a1bf8)

![image](https://github.com/user-attachments/assets/d5a02f71-15e2-4d74-a8c2-e87c569429ac)

For #159

WardBrian commented on gui/src/app/pages/HomePage/Sidebar.tsx:
```diff
@@ -62,7 +60,8 @@ const Sidebar: FunctionComponent<Sidebar> = ({
       <Toolbar />
 
-      <h3>Examples</h3>
+
+      <div className="SidebarContentWrapper SidebarHeading">Examples</div>
```
Is there a semantic reason to prefer (or not prefer) a styled div vs h3?

jsoules replied:
Technically there are accessibility reasons to prefer `H` tags where appropriate, but I didn't bring it up because a) we haven't been paying a lot of attention to that overall and b) it's not clear to me that it would actually be appropriate here anyway, as that sort of advice is really for a different sort of page/site...

WardBrian commented on gui/src/app/pages/HomePage/Sidebar.tsx:
```diff
@@ -72,43 +71,44 @@ const Sidebar: FunctionComponent<Sidebar> = ({
-          </Button>
-        </ListItem>
-      </List>
+      <div className="SidebarContentWrapper">
```
Similarly, it seems like actually rendering with `ul` (which is what the MUI ListItem renders as) might be more correct? If the margin is the issue we could place the entire list in this div

magland replied:
That's okay with me if you want to make that. I'm happy with how it actually renders as is, and I think in the future we'll want to revamp the sidebar to accommodate more advanced functionality.

magland commented:
@WardBrian @jsoules 

I went ahead and switched to using `<h3>` and `<ListItem>` again and updated the .css to achieve the margins, etc. that I was aiming for.

![image](https://github.com/user-attachments/assets/74c49d1a-4bac-43d1-bca4-fdc24ea27f4c)
